event_subscription.go

```diff
@@ -0,0 +1,99 @@
+package litefs
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+var (
+	EventSubscriptionClient = http.DefaultClient
+	EventSubscriptionURL    = "http://localhost:20202/events"
+)
+
+var (
+	errUnexpectedStatus = errors.New("unexpected status")
+)
+
+// EventSubscription tracks events published by a LiteFS node.
+type EventSubscription struct {
+	c     chan *Event
+	errc  chan error
+	ctx   context.Context
+	close func()
+}
```

Comment:
Having both the `EventSubscription` and `PrimaryMonitor` both use a goroutine seems like overkill. What do you think about having a simpler iterator-style `EventSubscription` instead:

```go
type Client struct

func (c *Client) SubscribeEvents(ctx context.Context) (*EventSubscription, error)


type EventSubscription struct

func (s *EventSubscription) Next() (*Event, error)
func (s *EventSubscription) Close() error
```

Reply:
I wrote it that way originally, but it made the logic of recovering from a failed/terminated request more challenging. I'll see if it feels less awkward with these being methods on a Client struct.

Reply:
This worked nicely.